spec/logger.spec.js

```diff
@@ -0,0 +1,57 @@
+describe("logger", function () {
+    "use strict";
+
+    var log, logger;
+
+    logger = require("../lib/logger");
+
+    beforeEach(function () {
+        spyOn(console, "log");
+        spyOn(console, "error");
+        spyOn(console.log, "apply");
+    });
+
+    describe("constructor", function () {
+        beforeEach(function () {
+            log = new logger({});
+        });
+
+        it("prints to log from info", function () {
+            log.info("something");
+            expect(console.log).toHaveBeenCalledWith("something");
+        });
+
+        it("prints to error from error", function () {
+            log.error("something");
+            expect(console.error).toHaveBeenCalledWith("ERROR: something");
+        });
+
+        it("prints to error from warn", function () {
+            log.warn("something");
+            expect(console.error).toHaveBeenCalledWith("WARN: something");
+        });
+
+        it("prints to log from console", function () {
+            log.console("something");
+            expect(console.log.apply.mostRecentCall.args[1]).toEqual(["something"]);
```

Comment:
If I don't spy on console.log.apply I can do expect(console.log).toHaveBeenCalledWith("something"). I'll go with this route.